src/fields_dump.cpp

```diff
@@ -0,0 +1,183 @@
+/* Copyright (C) 2005-2021 Massachusetts Institute of Technology
+%
+%  This program is free software; you can redistribute it and/or modify
+%  it under the terms of the GNU General Public License as published by
+%  the Free Software Foundation; either version 2, or (at your option)
+%  any later version.
+%
+%  This program is distributed in the hope that it will be useful,
+%  but WITHOUT ANY WARRANTY; without even the implied warranty of
+%  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+%  GNU General Public License for more details.
+%
+%  You should have received a copy of the GNU General Public License
+%  along with this program; if not, write to the Free Software Foundation,
+%  Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+*/
+
+// Dump/load raw fields data to/from an HDF5 file.  Only
+// works if the number of processors/chunks is the same.
+
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <cassert>
+
+#include "meep.hpp"
+#include "meep_internals.hpp"
+
+namespace meep {
+
+#define DUMP_FIELD(field_name)                                              \
+  {                                                                         \
+    /*                                                                      \
+     * make/save a num_chunks x NUM_FIELD_COMPONENTS x 2 array counting     \
+     * the number of entries in the 'field_name' array for each chunk.      \
+     */                                                                     \
+    size_t *num_f_ = new size_t[num_chunks * NUM_FIELD_COMPONENTS * 2];     \
```

Comment:
Can we please avoid C-style manual allocation in new code?

Reply:
done.